src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingSpec.cs

```diff
@@ -202,23 +239,96 @@ private void UpdateState(CounterChanged e)
             _count += e.Delta;
         }
     }
-    public class ClusterShardingNode1 : ClusterShardingSpec { }
-    public class ClusterShardingNode2 : ClusterShardingSpec { }
-    public class ClusterShardingNode3 : ClusterShardingSpec { }
+
+    internal class QualifiedCounter : Counter
+    {
+        public static Props Props(string typeName)
+        {
+            return Actor.Props.Create(() => new QualifiedCounter(typeName));
+        }
+
+        public readonly string TypeName;
+
+        public override string PersistenceId { get { return TypeName + "-" + Self.Path.Name; } }
+
+        public QualifiedCounter(string typeName)
+        {
+            TypeName = typeName;
+        }
+    }
+
+    internal class AnotherCounter : QualifiedCounter
+    {
+        public AnotherCounter()
+            : base("AnotherCounter")
+        {
+        }
+    }
+
+    internal class CounterSupervisor : ActorBase
+    {
+        public readonly IActorRef Counter;
+
+        public CounterSupervisor()
+        {
+            Counter = Context.ActorOf(Props.Create<Counter>(), "theCounter");
+        }
+
+        protected override SupervisorStrategy SupervisorStrategy()
+        {
+            return new AllForOneStrategy(Decider.From(ex =>
+            {
+                switch (ex)
+                {
+                    //case _: IllegalArgumentException     ⇒ SupervisorStrategy.Resume
+                    //case _: ActorInitializationException ⇒ SupervisorStrategy.Stop
+                    //case _: DeathPactException           ⇒ SupervisorStrategy.Stop
+                    //case _: Exception                    ⇒ SupervisorStrategy.Restart
+
+                    default:
+                        return Directive.Restart;
+                }
+            }));
+        }
+
+        protected override bool Receive(object message)
+        {
+            Counter.Forward(message);
+            return true;
+        }
+    }
+
+
+    public class ClusterShardingSpecRecoveryAll : ClusterShardingSpec
+    {
+        public ClusterShardingSpecRecoveryAll()
+            : this(new ClusterShardingSpecConfig(/*"all"*/))
+        {
+        }
+
+        protected ClusterShardingSpecRecoveryAll(ClusterShardingSpecConfig config)
+            : base(config)
+        {
+        }
+    }
+
+    //public class ClusterShardingSpecRecoveryConstant : ClusterShardingSpec
```

Comment:
didn't find a way yet how to run one spec against multiple configs.
Since the MultiNodeTestRunner is using Activator.CreateInstance to create new instance of config objects, it's not possible to have parameters on the ClusterShardingSpecConfig constructor. Is there a way around? I didn't want to duplicate whole config class.